unime/src-tauri/tests/tests/search_query.rs

```diff
@@ -12,95 +12,75 @@ async fn test_credential_search_query() {
     let state = json_example::<AppState>("tests/fixtures/states/two_credentials_redirect_me_query.json");
     let action = json_example::<Action>("tests/fixtures/actions/credential_search.json");
     let expected_state = json_example::<AppState>("tests/fixtures/states/two_credentials_search_query.json");
+
     assert_state_update(
-        AppStateContainer(Mutex::new(state)),
-        vec![action],
-        vec![Some(expected_state)],
+        AppStateContainer(Mutex::new(state.clone())),
+        vec![action.clone()],
+        vec![Some(expected_state.clone())],
     )
     .await;
 }
 
 #[tokio::test]
 #[serial_test::serial]
-async fn test_credential_sort_query() {
+async fn test_credential_add_recent_search() {
     setup_state_file();
     setup_stronghold();
 
-    let state = json_example::<AppState>("tests/fixtures/states/two_credentials_redirect_me_query.json");
-    let action = json_example::<Action>("tests/fixtures/actions/credential_sort.json");
-    let expected_state = json_example::<AppState>("tests/fixtures/states/two_credentials_sort_query.json");
+    // Add recent search with recent search still in the current field
+    let state = json_example::<AppState>("tests/fixtures/states/two_credentials_search_query.json");
+    let action = json_example::<Action>("tests/fixtures/actions/credential_add_recent_search.json");
+    let expected_state = json_example::<AppState>("tests/fixtures/states/two_credentials_recent_search.json");
     assert_state_update(
-        AppStateContainer(Mutex::new(state)),
+        AppStateContainer(Mutex::new(state.clone())),
         vec![action],
         vec![Some(expected_state)],
     )
     .await;
-}
-
-#[tokio::test]
-#[serial_test::serial]
-async fn test_credential_search_sort_query() {
-    setup_state_file();
-    setup_stronghold();
 
+    // Add recent search with recent search not in the current field
     let state = json_example::<AppState>("tests/fixtures/states/two_credentials_redirect_me_query.json");
-    let action = json_example::<Action>("tests/fixtures/actions/credential_search_sort.json");
-    let expected_state = json_example::<AppState>("tests/fixtures/states/two_credentials_search_sort_query.json");
+    let action = json_example::<Action>("tests/fixtures/actions/credential_add_recent_search.json");
+    let expected_state = json_example::<AppState>("tests/fixtures/states/two_credentials_recent_search_no_current.json");
     assert_state_update(
-        AppStateContainer(Mutex::new(state)),
+        AppStateContainer(Mutex::new(state.clone())),
         vec![action],
         vec![Some(expected_state)],
     )
     .await;
 }
 
+// To check wether searching the same search_hit isn't registered twice.
```

Comment:
Unnecessary comment, should be clear from the test name. Test name could be `test_add_existing_recent_search_does_not_create_duplicate()`.